pkg/storage/etcd/etcd_watcher.go

```diff
@@ -398,10 +404,14 @@ func (w *etcdWatcher) sendModify(res *etcd.Response) {
 	var oldObj runtime.Object
 	if res.PrevNode != nil && res.PrevNode.Value != "" {
 		// Ignore problems reading the old object.
-		if oldObj, err = w.decodeObject(res.PrevNode); err == nil {
+		if oldObj, err = w.uncachedDecodeObject(res.PrevNode); err == nil {
 			oldObjPasses = w.filter(oldObj)
 		}
+		if err := w.versioner.UpdateObject(oldObj, res.Node.ModifiedIndex); err != nil {
```

Comment:
Sorry, I was asking the wrong question.
I think it still makes sense to use the cache to get the deleted object. But we might want to delete it from the cache, right?